src/storm/modelchecker/blackbox/eMDP.h

```diff
@@ -2,19 +2,88 @@
 #define STORM_EMDP_H
 
 #include "storm/models/sparse/NondeterministicModel.h"
+#include "storm/storage/FlexibleSparseMatrix.h"
 
 namespace storm {
 namespace modelchecker {
 namespace blackbox {
 
+
+/*!
+* TODO
+*   - don't use FlexibleSparseMatrix as visitsMatrix -> memory inefficient because each row has to be filled to last entry
+*       -> Perhaps Vector of hashmaps is more reasonable
+*   - Make class child class of NondeterministicModel
+*       -> Problem: Nondeterministic model uses SparseMatrix
+*   - eMDP for now expects to know all states, that are processed.
+*       -> add checks or behavoir in case addVisit or getVisited was called with unknown states or actions
+*   - getVisited rename to getSampleCount
+*/
+template<class ValueType>
 class eMDP {
    public:
-    //addVisit
-    //getVisited
-    //addState
-    // print funktion
+    typedef uint_fast64_t index_type;
+    
+    /*!
+     * Constructs an empty eMDP
+     */
+    eMDP();
+
+    /*!
+     * increments the visits count of the given tripel by 1
+     * @param state   state index in which action was taken 
+     * @param action  action index of chosen action
+     * @param succ    state index of successor state
+     */
+    void addVisit(index_type state, index_type action, index_type succ);
```

Comment:
Es sollte klar sein, ob hier und bei den anderen Funktionen der interne index oder der externe index benutzt wird (vor dem mapping mithilfe von explorationOrder oder danach)

Reply:
Von außen sollte man keinen Zugriff auf das interne Mapping haben. Sprich: man kann von außen nur das externe Mapping verwenden.
Ich denke das interne Mapping wird aus eMDP verschwinden sobald @maximiliankamps die neue Speicherstruktur fertig hat.
Dieses Problem sollte sich daher in gegebener Zeit von selbst klären